parser.y

```diff
@@ -8455,11 +8455,12 @@ CreateRoleStmt:
 
 /* See http://dev.mysql.com/doc/refman/5.7/en/alter-user.html */
 AlterUserStmt:
-	"ALTER" "USER" IfExists UserSpecList
+	"ALTER" "USER" IfExists UserSpecList RequireClause ConnectionOptions PasswordOrLockOptions
 	{
 		$$ = &ast.AlterUserStmt{
 			IfExists: $3.(bool),
 			Specs: $4.([]*ast.UserSpec),
+			PasswordOrLockOptions: $7.([]*ast.PasswordOrLockOption),
```

Comment:
add the following fields for AlterUserStmt please, and add test cases for it:
```
	TslOptions            []*TslOption
	ResourceOptions       []*ResourceOption
```

Reply:
for `RequireClause`、`ConnectionOptions` and `PasswordOrLockOptions` stmts, TiDB is not supporting.
could you add warns for this stmts?

Reply:
both create and alter stmts?

Reply:
yes

Reply:
PTAL